autotest/utilities/test_gdalalg_raster_clip.py

```diff
@@ -0,0 +1,310 @@
+#!/usr/bin/env pytest
+# -*- coding: utf-8 -*-
+###############################################################################
+# Project:  GDAL/OGR Test Suite
+# Purpose:  'gdal raster clip' testing
+# Author:   Even Rouault <even dot rouault @ spatialys.com>
+#
+###############################################################################
+# Copyright (c) 2024, Even Rouault <even dot rouault at spatialys.com>
+#
+# SPDX-License-Identifier: MIT
+###############################################################################
+
+import gdaltest
+import pytest
+
+from osgeo import gdal
+
+
+def get_alg():
+    return gdal.GetGlobalAlgorithmRegistry()["raster"]["clip"]
+
+
+def test_gdalalg_raster_clip_missing_bbox_or_like():
+
+    alg = get_alg()
+    alg["input"] = "../gcore/data/byte.tif"
+    alg["output"] = ""
+    alg["output-format"] = "MEM"
+    with pytest.raises(
+        Exception, match="clip: --bbox, --geometry or --like must be specified"
+    ):
+        alg.Run()
+
+
+def test_gdalalg_raster_clip_input_error(tmp_vsimem):
+
+    alg = get_alg()
+    alg["input"] = gdal.GetDriverByName("MEM").Create("", 1, 1)
+    alg["output"] = ""
+    alg["output-format"] = "MEM"
+    alg["bbox"] = [440780, 3750200, 441860, 3751260]
+    with pytest.raises(
+        Exception, match="Clipping is not supported on a raster without a geotransform"
+    ):
+        alg.Run()
+
+    src_ds = gdal.GetDriverByName("MEM").Create("", 1, 1)
+    src_ds.SetGeoTransform([2, 1, -1, 49, -1, -1])
+
+    alg = get_alg()
+    alg["input"] = src_ds
+    alg["output"] = ""
+    alg["output-format"] = "MEM"
+    alg["bbox"] = [440780, 3750200, 441860, 3751260]
+    with pytest.raises(
+        Exception,
+        match="Clipping is not supported on a raster whose geotransform has rotation terms",
+    ):
+        alg.Run()
+
+
+def test_gdalalg_raster_clip_bbox_and_like():
+
+    alg = get_alg()
+    alg["input"] = "../gcore/data/byte.tif"
+    alg["output"] = ""
+    alg["output-format"] = "MEM"
+    alg["bbox"] = [440780, 3750200, 441860, 3751260]
+    alg.Run()
+    ds = alg["output"].GetDataset()
+    assert ds.RasterXSize == 18
+    assert ds.RasterYSize == 18
+    assert ds.GetSpatialRef().GetAuthorityCode(None) == "26711"
+    assert ds.GetGeoTransform() == pytest.approx(
+        (440780, 60, 0, 3751260, 0, -60), rel=1e-8
+    )
+    assert ds.GetRasterBand(1).Checksum() == 3695
+
+    alg = get_alg()
+    alg["input"] = "../gcore/data/byte.tif"
+    alg["output"] = ""
+    alg["output-format"] = "MEM"
+    alg["like"] = ds
+    alg.Run()
+    ds = alg["output"].GetDataset()
+    assert ds.RasterXSize == 18
+    assert ds.RasterYSize == 18
+    assert ds.GetSpatialRef().GetAuthorityCode(None) == "26711"
+    assert ds.GetGeoTransform() == pytest.approx(
+        (440780, 60, 0, 3751260, 0, -60), rel=1e-8
+    )
+    assert ds.GetRasterBand(1).Checksum() == 3695
+
+
```

Comment:
Add test for "like" with rotated raster?

Reply:
I'll let that aside